pkg/operator/configobservation/scheduler/observe_scheduler.go

```diff
@@ -10,6 +11,8 @@ import (
 	"k8s.io/klog"
 )
 
+const workerNodeSelector = "node-role.kubernetes.io/worker=\"\""
```

Comment:
This is not guaranteed to exist in all clusters.

Node roles are not an api.

Reply:
IIUC, there are 2 ways to distinguish master nodes from worker nodes:

- role label
- Master taints

So, are you suggesting, we cannot rely on role label which the `defaultNodeSelector` is based of? IIRC, we used to have something like `type=Compute` as defaultNodeSelector in 3.11. 

This is not to say that we cannot have a validation against pods have master tolerations

Reply:
@smarterclayton Can you explain what you mean by "Node roles are not an api and no business logic should based on node roles."? It has been common practice with OpenShift in the field to add role labels to nodes and separate workload based on these labels by using node selectors and the `openshift.io/node-selector` annotation. This has been a very effective design pattern, what is inappropriate about this? Should we encourage use of a different label?